library/src/test/java/com/novoda/downloadmanager/DownloadManagerTest.java

```diff
@@ -0,0 +1,239 @@
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willAnswer;
+import static org.mockito.Mockito.*;
+
+public class DownloadManagerTest {
+
+    private static final InternalDownloadBatchStatus BATCH_STATUS = anInternalDownloadsBatchStatus().build();
+    private static final InternalDownloadBatchStatus ADDITIONAL_BATCH_STATUS = anInternalDownloadsBatchStatus().build();
+    private static final DownloadBatchId DOWNLOAD_BATCH_ID = aDownloadBatchId().withRawDownloadBatchId("id01").build();
+    private static final DownloadBatchId ADDITIONAL_DOWNLOAD_BATCH_ID = aDownloadBatchId().withRawDownloadBatchId("id02").build();
+    private static final Batch BATCH = new Batch.Builder(DOWNLOAD_BATCH_ID, "title").build();
+
+    private final AllStoredDownloadsSubmittedCallback allStoredDownloadsSubmittedCallback = mock(AllStoredDownloadsSubmittedCallback.class);
+    private final AllBatchStatusesCallback allBatchStatusesCallback = mock(AllBatchStatusesCallback.class);
+    private final DownloadService downloadService = mock(DownloadService.class);
+    private final Object lock = mock(Object.class);
+    private final DownloadBatch downloadBatch = mock(DownloadBatch.class);
+    private final DownloadBatch additionalDownloadBatch = mock(DownloadBatch.class);
+    private final ExecutorService executorService = mock(ExecutorService.class);
+    private final DownloadBatchCallback downloadBatchCallback = mock(DownloadBatchCallback.class);
+    private final FileOperations fileOperations = mock(FileOperations.class);
+    private final DownloadsBatchPersistence downloadsBatchPersistence = mock(DownloadsBatchPersistence.class);
+    private final LiteDownloadManagerDownloader downloadManagerDownloader = mock(LiteDownloadManagerDownloader.class);
+
+    private DownloadManager downloadManager;
+    private Map<DownloadBatchId, DownloadBatch> downloadBatches = new HashMap<>();
+    private List<DownloadBatchStatus> downloadBatchStatuses = new ArrayList<>();
+    private List<DownloadBatchCallback> downloadBatchCallbacks = new ArrayList<>();
+
+    @Before
+    public void setUp() {
+        downloadBatches = new HashMap<>();
+        downloadBatches.put(DOWNLOAD_BATCH_ID, downloadBatch);
+        downloadBatches.put(ADDITIONAL_DOWNLOAD_BATCH_ID, additionalDownloadBatch);
+
+        downloadBatchCallbacks.add(downloadBatchCallback);
+
+        downloadManager = new DownloadManager(
+                lock,
+                executorService,
+                downloadBatches,
+                downloadBatchCallbacks,
+                fileOperations,
+                downloadsBatchPersistence,
+                downloadManagerDownloader
+        );
+
+        final ArgumentCaptor<DownloadsBatchPersistence.LoadBatchesCallback> loadBatchesCallbackCaptor = ArgumentCaptor.forClass(DownloadsBatchPersistence.LoadBatchesCallback.class);
+        willAnswer(new Answer<Void>() {
```

Comment:
And do it when needed rather than on set up

Reply:
I don't see the benefit of moving this @pablisco this is essentially setting up the mock behaviour that will be used in the tests. I can move to the test but then it will make that test much more difficult to read.